openpype/hosts/nuke/plugins/publish/validate_asset_context.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+"""Validate if instance asset is the same as context asset."""
+from __future__ import absolute_import
+
+import pyblish.api
+
+import openpype.hosts.nuke.api.lib as nlib
+
+from openpype.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishXmlValidationError,
+    OptionalPyblishPluginMixin,
+    get_errored_instances_from_context
+)
+
+
+class SelectInvalidNodesAction(pyblish.api.Action):
+    """Select invalid nodes."""
+
+    label = "Select Failed Node"
+    icon = "briefcase"
+    on = "failed"
+
+    def process(self, context, plugin):
+        if not hasattr(plugin, "select"):
+            raise RuntimeError("Plug-in does not have repair method.")
+
+        # Get the failed instances
+        self.log.debug("Finding failed plug-ins..")
+        failed_instance = get_errored_instances_from_context(context, plugin)
+        if failed_instance:
+            self.log.debug("Attempting selection ...")
+            plugin.select(failed_instance.pop())
+
+
+class ValidateCorrectAssetContext(
+    pyblish.api.InstancePlugin,
+    OptionalPyblishPluginMixin
+):
+    """Validator to check if instance asset context match context asset.
+
+    When working in per-shot style you always publish data in context of
+    current asset (shot). This validator checks if this is so. It is optional
+    so it can be disabled when needed.
+
+    Checking `asset` and `task` keys.
+    """
+    order = ValidateContentsOrder
+    label = "Validate asset context"
+    hosts = ["nuke"]
+    actions = [
+        RepairAction,
+        SelectInvalidNodesAction,
+    ]
+    optional = True
+
+    @classmethod
+    def apply_settings(cls, project_settings):
+        """Apply deprecated settings from project settings.
+        """
+        nuke_publish = project_settings["nuke"]["publish"]
+        if "ValidateCorrectAssetName" not in nuke_publish:
+            return
+
+        deprecated_setting = nuke_publish["ValidateCorrectAssetName"]
+        cls.enabled = deprecated_setting["enabled"]
+        cls.optional = deprecated_setting["optional"]
+        cls.active = deprecated_setting["active"]
```

Comment:
This won't work - by defining `apply_settings` you're basically saying "always apply settings like this". And with the logic here it will never apply the _new_ settings.

Reply:
yeah you are right, this needs to be fixed. 4c90065f43930c35e1ad37712b0ec74d49a0b4d2